service/src/main/java/greencity/service/locations/LocationApiService.java

```diff
@@ -0,0 +1,355 @@
+        } catch (NotFoundException e) {
+            allRegions = getAllRegions();
+        }
+        if (allRegions.isEmpty()) {
+            allRegions = getAllRegions();
+        }
+        return allRegions.stream()
+            .filter(region -> region.getName().containsKey(regionName) || region.getName()
+                .containsValue(regionName))
+            .findFirst()
+            .orElseThrow(() -> new NotFoundException(ErrorMessage.REGION_NOT_FOUND + regionName));
+    }
+
+    /**
+     * Retrieves a city in a specific region.
+     *
+     * @param regionName The name of the region.
+     * @param cities     A list of LocationDto objects representing cities.
+     * @return A LocationDto object containing the city's data.
+     */
+    public LocationDto getCityInRegion(String regionName, List<LocationDto> cities) {
+        LocationDto region = getRegionByName(regionName);
+        String regionID = region.getId();
+        for (LocationDto city : cities) {
+            LocationDto localCommunity = (getLocationDataByCode(2, 3, city.getParentId()));
+            LocationDto districtRegion = (getLocationDataByCode(2, 2, localCommunity.getParentId()));
+            if (districtRegion.getParentId().equals(regionID)) {
+                return city;
+            }
+        }
+        throw new NotFoundException(ErrorMessage.CITY_NOT_FOUND + cities.get(0).getName());
+    }
+
+    /**
+     * Retrieves a city by its name from the regional side.
+     *
+     * @param regionName The name of the region where the city is located.
+     * @param cityName   The name of the city.
+     * @return A LocationDto object containing the city's data.
+     */
+    public LocationDto getCityByNameFromRegionSide(String regionName, String cityName) {
+        LocationDto region = getRegionByName(regionName);
+        List<LocationDto> districts = getAllDistrictInTheRegionsById(region.getId());
+        List<LocationDto> localCommunities = districts.stream()
+            .flatMap(district -> getAllLocalCommunitiesById(district.getId()).stream())
+            .collect(Collectors.toList());
+        List<LocationDto> cities = localCommunities.stream()
+            .flatMap(community -> getAllCitiesById(community.getId()).stream())
+            .collect(Collectors.toList());
+        return findLocationByName(cities, regionName, cityName, ErrorMessage.CITY_NOT_FOUND);
+    }
+
+    /**
+     * Retrieves a city in a specific region by its name.
+     *
+     * @param regionName The name of the region where the city is located.
+     * @param cityName   The name of the city.
+     * @return A LocationDto object containing the city's data.
+     */
+    public List<LocationDto> getAllDistrictsInCityByNames(String regionName, String cityName) {
+        if (cityName.equals("Київ") || cityName.equals("Kyiv")) {
+            return getAllDistrictsInCityByCityID("UA80000000000093317");
+        }
+        List<LocationDto> allDistricts = null;
```

Comment:
Any reason you set `allDistricts = null`?
we can use 
`List<LocationDto> allDistricts = getAllDistrictsInCityByCityID(cityId);`